src/test/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PubsubTest.kt

```diff
@@ -15,40 +15,65 @@
  */
 package com.bardsoftware.backend.fleet.rmarkdown
 
+import com.google.protobuf.ByteString
+import com.google.pubsub.v1.PubsubMessage
+import org.junit.After
 import org.junit.Assert.assertEquals
+import org.junit.Before
 import org.junit.Test
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.nio.file.Files
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
 
 class PubsubTest {
-    @Test
-    fun testSimpleMessage() {
-        val message = "hello"
-        val sum = "b1946ac92492d2347c6235b4d2611184  -\n"
+    val tasksDir = "tasks"
 
-        val testCallback = { acceptedMessage: String, acceptedMd5sum: String ->
-            assertEquals(message, acceptedMessage)
-            assertEquals(sum, acceptedMd5sum)
-        }
 
-        val manager = SubscribeManager("", testCallback)
-        manager.pushMessage(message)
+    @Before
+    fun createDir() {
+        File(tasksDir).mkdir()
     }
 
     @Test
-    fun testMultipleMessages() {
-        val message = "hello"
-        val sum = "b1946ac92492d2347c6235b4d2611184  -\n"
-
-        var messagesCount = 0
-        val testCallback = { acceptedMessage: String, acceptedMd5sum: String ->
-            messagesCount++
-            assertEquals(message, acceptedMessage)
-            assertEquals(sum, acceptedMd5sum)
+    fun processFileUnzipTest() {
```

Comment:
You may want to add tests for bad cases, like directory doesn't exist or not writable or Zip file is malformed, etc.

Reply:
Added